### PR TITLE
Add spotlight rankings

### LINKED ISSUE #1169: Spotlight rankings
<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/45323393-spotlight-rankings?utm_campaign=plugin&utm_content=tracker%2F21853994&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F21853994&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by notbakaneko**
closes #1169

- add spotlight (`chart`) rankings
- some `RankingController` reorganization
- reverted back to single page for all the chart rankings until other changes to how spotlights and some other stuff are handled are more finalized

---

### CODE REVIEW

**Comment by notbakaneko** [comment]
still deciding on whether the composed `SelectOptions` should have different block names or modifiers instead 💭

**Comment by nekodex** [comment]
this is blocked waiting on country filtering removal

File: config/database.php [hunk 1]
```diff
@@ -90,6 +90,22 @@
             ],
         ],
 
+        'mysql-charts' => [
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply config/database.php]: should be done separately

File: app/Http/Controllers/RankingController.php [hunk 2]
```diff
@@ -23,34 +23,74 @@
...
+            if ($type === null) {
+                return ujs_redirect(route('rankings', ['mode' => $mode, 'type' => 'performance']));
+            }
+
+            if (!in_array($type, static::RANKING_TYPES, true)) {
+                abort(404);
+            }
+
+            if (request()->has('country') && !in_array($type, static::SPOTLIGHT_TYPES, true)) {
+                $countryStats = CountryStatistics::where('display', 1)
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply app/Http/Controllers/RankingController.php]: not very relevant to the PR

File: resources/views/rankings/_spotlight_rankings_table.blade.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    #{{ $index + 1 }}
+                </td>
+                <td class="ranking-page-table__column">
+                    <div class="ranking-page-table__user-link">
+                        @include('objects._country_flag', [
+                            'country_name' => $score->user->country->name,
+                            'country_code' => $score->user->country->acronym,
+                        ])
+                        <a href="{{route('users.show', $score->user_id)}}">
+                            <span class="ranking-page-table__user-link-text js-usercard" data-user-id="{{$score->user_id}}" data-tooltip-position="right center">
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/views/rankings/_spotlight_rankings_table.blade.php]: not very relevant to the PR

**Pull request merged**